publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java

```diff
@@ -136,4 +158,34 @@ private boolean isThesis(CreatePublicationRequest request) {
 
         return requestInstanceType.isPresent() && THESIS_INSTANCE_TYPES.contains(requestInstanceType.get());
     }
+
+    private CustomerAwareUserContext customerAwareUserContextFromExternalClient(
+        final RequestInfo requestInfo,
+        final IdentityServiceClient identityServiceClient) throws UnauthorizedException {
+
+        var client = attempt(() -> requestInfo.getClientId().orElseThrow())
+                         .map(identityServiceClient::getExternalClient)
+                         .orElseThrow(fail -> new UnauthorizedException());
+
+        final var customerUri = client.getCustomerUri();
+        var resourceOwner = new ResourceOwner(
+            new Username(client.getActingUser()),
+            client.getCristinUrgUri()
+        );
+
+        final var userInstance = UserInstance.createExternalUser(resourceOwner, client.getCustomerUri());
+
+        return new CustomerAwareUserContext(userInstance, customerUri);
+    }
+
+    private CustomerAwareUserContext customerAwareUserContextFromInternalUser(RequestInfo requestInfo)
+        throws UnauthorizedException {
+        var resourceOwner = createInternalResourceOwner(requestInfo);
+        var customerUri = requestInfo.getCurrentCustomer();
+        return new CustomerAwareUserContext(UserInstance.create(resourceOwner, customerUri), customerUri);
```

Comment:
Instead of sending customerUri two times, maybe consider something like this:
```

private CustomerAwareUserContext customerAwareUserContextFromInternalUser(RequestInfo requestInfo)
        throws UnauthorizedException {
        var resourceOwner = createInternalResourceOwner(requestInfo);
        var userInstance = UserInstance.create(resourceOwner, requestInfo.getCurrentCustomer());
        return CustomerAwareUserContext.fromUserInstance(userInstance);
    }
        
        public static CustomerAwareUserContext fromUserInstance(UserInstance userInstance) {
            return new CustomerAwareUserContext(userInstance, userInstance.getOrganizationUri());
        }
```